tests/test_optimizers.py

```diff
@@ -13,8 +14,12 @@
 ]
 
 
-@pytest.mark.parametrize("tm", tm)
+@pytest.mark.parametrize("tm", test_matrix)
 def test_get_weights(tm):
+    stretched_component_gram_matrix = tm[0][0]
+    linear_coefficient = tm[0][1]
+    lower_bound = tm[0][2]
+    upper_bound = tm[0][3]
     expected = tm[1]
-    actual = get_weights(tm[0][0], tm[0][1], tm[0][2], tm[0][3])
+    actual = get_weights(stretched_component_gram_matrix, linear_coefficient, lower_bound, upper_bound)
```

Comment:
pass variables instead of `tm[0][0]`, etc. which is much harder to read for future generations.